aws/resource_aws_resource_group.go

```diff
@@ -0,0 +1,169 @@
+						"query": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  resourcegroups.QueryTypeTagFilters10,
+							ValidateFunc: validation.StringInSlice([]string{
+								resourcegroups.QueryTypeTagFilters10,
+							}, false),
+						},
+					},
+				},
+			},
+
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func extractResourceGroupResourceQuery(resourceQueryList []interface{}) *resourcegroups.ResourceQuery {
+	resourceQuery := resourceQueryList[0].(map[string]interface{})
+
+	return &resourcegroups.ResourceQuery{
+		Query: aws.String(resourceQuery["query"].(string)),
+		Type:  aws.String(resourceQuery["type"].(string)),
+	}
+}
+
+func resourceAwsResourceGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).resourcegroupsconn
+
+	input := resourcegroups.CreateGroupInput{
+		Description:   aws.String(d.Get("description").(string)),
+		Name:          aws.String(d.Get("name").(string)),
+		ResourceQuery: extractResourceGroupResourceQuery(d.Get("resource_query").(*schema.Set).List()),
+	}
+
+	res, err := conn.CreateGroup(&input)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(aws.StringValue(res.Group.Name))
+
+	return resourceAwsResourceGroupRead(d, meta)
+}
+
+func resourceAwsResourceGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).resourcegroupsconn
+
+	g, err := conn.GetGroup(&resourcegroups.GetGroupInput{
+		GroupName: aws.String(d.Id()),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	d.Set("name", aws.StringValue(g.Group.Name))
+	d.Set("description", aws.StringValue(g.Group.Description))
+	d.Set("arn", aws.StringValue(g.Group.GroupArn))
+
+	q, err := conn.GetGroupQuery(&resourcegroups.GetGroupQueryInput{
+		GroupName: aws.String(d.Id()),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	resultQuery := map[string]interface{}{}
+	resultQuery["query"] = aws.StringValue(q.GroupQuery.ResourceQuery.Query)
+	resultQuery["type"] = aws.StringValue(q.GroupQuery.ResourceQuery.Type)
+	d.Set("resource_query", []map[string]interface{}{resultQuery})
+
+	return nil
+}
+
+func resourceAwsResourceGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).resourcegroupsconn
+
+	if d.HasChange("description") {
+		input := resourcegroups.UpdateGroupInput{
+			GroupName:   aws.String(d.Id()),
+			Description: aws.String(d.Get("description").(string)),
+		}
+
+		_, err := conn.UpdateGroup(&input)
+		if err != nil {
+			return err
+		}
+	}
+
+	if d.HasChange("resource_query") {
+		input := resourcegroups.UpdateGroupQueryInput{
+			GroupName:     aws.String(d.Id()),
+			ResourceQuery: extractResourceGroupResourceQuery(d.Get("resource_query").([]interface{})),
+		}
+
+		_, err := conn.UpdateGroupQuery(&input)
+		if err != nil {
+			return err
```

Comment:
Nit: Same here, regarding context. 

```suggestion
			return fmt.Errorf("error updating Resource Groups Group (%s) Query: %s", d.Id(), err)
```